woodwork/utils.py

```diff
@@ -298,3 +298,112 @@ def _parse_logical_type(logical_type, name):
+    will not allow duplicate columns at concatenation.
+
+    Args:
+        objs (list[Series, DataFrame]): The Woodwork objects to be concatenated. If Woodwork
+            is not initialized on any of the objects, type inference will be performed.
+        validate_schema (bool, optional): Whether validation should be performed on the typing information
+            for the concatenated DataFrame. Defaults to True.
+
+    Returns:
+        DataFrame: A Woodwork dataframe whose typing information is also a concatenation of the input dataframes.
+    """
+    if not objs:
+        raise ValueError('No objects to concatenate')
+
+    table_name = ''
+    logical_types = {}
+    semantic_tags = {}
+    col_descriptions = {}
+    col_metadata = {}
+    table_metadata = {}
+    use_standard_tags = {}
+
+    index = None
+    time_index = None
+
+    concat_objs = []
+    # Record the typing information for all the columns that have Woodwork schemas
+    col_names_seen = set()
+    for obj in objs:
+        ww_columns = {}
+        if isinstance(obj.ww.schema, ww.table_schema.TableSchema):
+            # Raise error if there's overlap between table metadata
+            overlapping_keys = obj.ww.metadata.keys() & table_metadata.keys()
+            if overlapping_keys:
+                raise ValueError(f'Cannot resolve overlapping keys in table metadata: {overlapping_keys}')
+
+            table_metadata = {**obj.ww.metadata, **table_metadata}
+
+            # Combine table names
+            if obj.ww.name is not None:
+                if table_name:
+                    table_name += '_'
+                table_name += str(obj.ww.name)
+
+            # Cannot have multiple tables with indexes or time indexes set
+            for idx_type, obj_idx, table_idx in [('index', obj.ww.index, index),
+                                                 ('time_index', obj.ww.time_index, time_index)]:
+                if obj_idx is not None:
+                    if table_idx is None:
+                        if idx_type == 'index':
+                            index = obj_idx
+                        elif idx_type == 'time_index':
+                            time_index = obj_idx
+                    else:
+                        raise IndexError(f'Cannot have {idx_type} columns set on more than one table in objs. '
+                                         f'Please remove the {idx_type} columns from all but one table.')
+
+            ww_columns = obj.ww.schema.columns
+        elif isinstance(obj.ww.schema, ww.column_schema.ColumnSchema):
+            ww_columns = {obj.name: obj.ww.schema}
+
+        concat_objs.append(obj)
+
+        # Compile the typing information per column
+        for name, col_schema in ww_columns.items():
+            # Do not look at typing information for duplicate columns.
+            # Means that index columns will only have first occurrance's typing information
+            if name in col_names_seen:
+                continue
```

Comment:
If I understand right, if we have duplicate column names, we keep the first one we encounter and ignore the rest. I'm wondering if it would be better to raise an error instead of continue here so the user has an opportunity to resolve.

Reply:
Upon further review, it seems the in the case of duplicate column names we get an error below when we try to init Woodwork. Am I understanding correctly? If so, I wonder if it would be better to go ahead and raise that error here instead of waiting for init? Also, I think the comment above is no longer valid if we error when duplicate cols are present.

Reply:
So the comment above these lines isn't relevant no matter what. But I'm not opposed to just catching the duplicate columns there. Plus, and this just occured to me, we can catch duplicate index columns under the general duplicate column check, which would greatly simplify things, but would mean we'd lose some of the specificity in the error message. Though, the instructions are the same whether or not the column is an index.

An idea for an error message:

> Duplicate column 'id' has been found in more than one input object. Please remove duplicate columns from all but one table.

Reply:
I think the loss of specificity in the error message is worth it to simplify the code. I also think the error message you have suggested sounds good!

Reply:
I might have been overly optimistic here. We still need to be able to catch the case where the index columns might have different names, so some of the logic will have to stay, but we can catch that as a separate case, and I'll separate the index and time index logic os it's hopefully less confusing

Reply:
@thehomebrewnerd Made these changes. Let me know if the duplicate column logic and index logic make sense to you!

Reply:
I like it! Much easier for my simple mind to follow!